flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/declarative/WaitingForResourcesTest.java

```diff
@@ -0,0 +1,328 @@
+                    .build();
+        }
+
+        @Override
+        public boolean hasEnoughResources(ResourceCounter desiredResources) {
+            return hasEnoughResourcesSupplier.get();
+        }
+
+        @Override
+        public ExecutionGraph createExecutionGraphWithAvailableResources() throws Exception {
+            return createExecutionGraphWithAvailableResources.get();
+        }
+
+        @Override
+        public void runIfState(State expectedState, Runnable action, Duration delay) {
+            actions.add(Tuple3.of(expectedState, action, delay));
+        }
+
+        // ---- Testing extensions ------
+
+        public void setWaitingForResources(WaitingForResources wfr) {
+            this.waitingForResources = wfr;
+        }
+
+        public List<Tuple3<State, Runnable, Duration>> getActions() {
+            return actions;
+        }
+
+        public void setHasEnoughResources(Supplier<Boolean> sup) {
+            hasEnoughResourcesSupplier = sup;
+        }
+
+        public void setCreateExecutionGraphWithAvailableResources(Supplier<ExecutionGraph> sup) {
+            this.createExecutionGraphWithAvailableResources = sup;
+        }
+
+        private class MockExecutingContext implements Executing.Context {
+            @Override
+            public void goToCanceling(
+                    ExecutionGraph executionGraph,
+                    ExecutionGraphHandler executionGraphHandler,
+                    OperatorCoordinatorHandler operatorCoordinatorHandler) {}
+
+            @Override
+            public Executing.FailureResult howToHandleFailure(Throwable failure) {
+                return null;
+            }
+
+            @Override
+            public boolean canScaleUp(ExecutionGraph executionGraph) {
+                return false;
+            }
+
+            @Override
+            public void goToRestarting(
+                    ExecutionGraph executionGraph,
+                    ExecutionGraphHandler executionGraphHandler,
+                    OperatorCoordinatorHandler operatorCoordinatorHandler,
+                    Duration backoffTime) {}
+
+            @Override
+            public void goToFailing(
+                    ExecutionGraph executionGraph,
+                    ExecutionGraphHandler executionGraphHandler,
+                    OperatorCoordinatorHandler operatorCoordinatorHandler,
+                    Throwable failureCause) {}
+
+            @Override
+            public void runIfState(State expectedState, Runnable action) {}
+
+            @Override
+            public boolean isState(State expectedState) {
+                return false;
+            }
+
+            @Override
+            public Executor getMainThreadExecutor() {
+                return ForkJoinPool.commonPool();
+            }
+
+            @Override
+            public void goToFinished(ArchivedExecutionGraph archivedExecutionGraph) {}
+        }
+    }
```

Comment:
This seems wrong. We shouldn't have to implement something for the `Executing` state when testing the `WaitingForResources` test.